src/ActionList/Divider.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react'
+import styled from 'styled-components'
+import {get} from '../constants'
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface DividerProps {}
+
+const StyledDivider = styled.div`
+  position: relative;
+  height: 1px;
+  background: ${get('colors.selectMenu.borderSecondary')};
+  border: 0;
+  margin: calc(${get('space.2')} - 1px) -${get('space.2')} ${get('space.2')};
```

Comment:
Negative margins always feel like an antipattern to me, and the `- 1px` seems a bit hacky. Can the style be cleaned up?

Reply:
This is adapted from [primer/overlay-prototype actionList.scss#L391](https://github.com/primer/overlay-prototype/blob/0b06227bd0ef15fa1e1ff967c01ce6289fe358f6/src/scss/actionList.scss#L391). @T-Hugs, do you have a suggested alternative? cc @vdepizzol

Reply:
Are we beholden to Primer CSS in any way? I can't offer an edit suggestion unless I clone and look at all related components together, which I am happy to do.

Reply:
For context, some of the `calc`ulations (e.g. https://github.com/primer/components/blob/f95e1a0aa9189ced510cd408bf1e6bff83c6a90c/src/ActionList/Header.tsx#L12) are here because [`primer/overlay-prototype` uses](https://github.com/primer/overlay-prototype/blob/0b06227bd0ef15fa1e1ff967c01ce6289fe358f6/src/scss/variables.scss#L26-L41) the proposed [4px grid spacing](https://github.com/github/primer/issues/41).

Reply:
@smockle @T-Hugs the other option would be to wrap the `ActionList-item` elements in an extra element that would contain the inset padding, that one being a sibling of the divider itself... I don't know if this something along these lines is feasible or makes sense:
```
<ActionList>
  <insetContainer>
    <ActionList.item/>
    <ActionList.item/>
  </insetContainer>
  <Divider>
  <insetContainer>
    <ActionList.item/>
  </insetContainer>
</ActionList>
```

I think adding margins directly to `ActionList items` is troublesome, that's why I haven't gone that route (and had to adopt negative margins to dividers), but maybe I'm overthinking 🤔

Reply:
I was originally concerned with the margins being set individually in the row items while thinking that we in the future we might also have different mobile-specific styles, like full-width rows... but I guess that's not correlated. 🤔 

@T-Hugs I think what you're saying makes sense: actionList containers could have vertical padding (8px top and bottom), with actionList items themselves having horizontal margins (8px left and right).

@smockle can we include this as part of a `style: inset` property of ActionList? We only have that one for now but we might need some flexibility in the future :)

Reply:
Fixed in https://github.com/primer/components/pull/1078/commits/d70271b8beea674305fa8359de6c68f6742d9d05!

That commit shuffles spacing styles but is visually a no-op:
* Omnidirectional `padding` is removed from `ErsatzOverlay` so its contents are displayed edge-to-edge
* Negative horizontal `margin` is removed from `Divider`, which continues to be displayed edge-to-edge. `Divider`’s positive vertical `margin` is retained.
* Positive horizontal `margin` is added to `Header` and `Item`, so that they appear indented relative to `Divider`.
* Horizontal padding is added to `List`, so the first and last `Items` are offset from its top and bottom edges.

Reply:
@vdepizzol If `ActionList` has a `style` _other than_ `"inset"`, should I remove `List`’s top and bottom padding _and_ also remove the left and right margin from each `Header` and `Item` it contains?

Reply:
@smockle yes, `List`'s vertical padding and `Item`/`Header` horizontal padding should be tied to an `inset` `ActionList` `style`.

In the future, I see a `fullWidth` style which doesn't have any inset paddings around items, but have the items covering the full width of the container where they exist. Think of an iOS [UITableView](https://developer.apple.com/documentation/uikit/uitableview) equivalent for local navigation etc.

Reply:
Added support for list style variants in https://github.com/primer/components/pull/1078/commits/2ca940242ed28a55081d24ec68ba46e25a12cffc!